File: api/BeSureApi/Models/ImprestStock.cs
using BeSureApi.Models;

namespace BeSureApi.Models
{
    public class ImprestStock
    {
        public int ContractId {  get; set; }
        public int CustomerId {  get; set; }
        public int? CustomerSiteId {  get; set; }
        public int? ServiceEngineerId { get; set; }
        public int[] PartStockIdList {  get; set; }
        public string? Remarks {  get; set; }
        public string ReservedFrom {  get; set; }
        public string ReservedTo {  get; set; }
        public bool IsCustomerSite { get; set; }
        public string IsbyCourier { get; set;}
    }

    public class ImprestStockCreate
    {
        public ImprestStock impreststock { get; set; }
        public DeliveryChallan? deliverychallan { get; set; }
    }
}

File: ui/src/types/impreststock.ts
import { Decoder, boolean, object } from "decoders";

export interface ImprestStock {
    PartStockIdList:number[];
    CustomerId: number;
    ContractId: number;
    CustomerSiteId: number | null;
    ServiceEngineerId: number | null;
    ReservedFrom: string;
    ReservedTo: string;
    Remarks:string;
    IsCustomerSite:boolean;
    IsbyCourier: string;
}

export interface ImprestStockResponse {
    IsImprestStockCreated: Boolean;
}

export const imprestStockResponseDecoder: Decoder<ImprestStockResponse> = object({
    IsImprestStockCreated: boolean,
});